packages/next/src/build/webpack-config.ts

```diff
@@ -1313,20 +1274,27 @@ export default async function getBaseWebpackConfig(
       request === 'react/jsx-runtime'
     ) {
       if (isAppLayer) {
-        return `commonjs next/dist/compiled/${request.replace(
-          'react',
-          'react' + bundledReactChannel
-        )}`
+        return `commonjs ${request}`
       }
       return
     }
 
-    // Special internal modules that must be bundled for Server Components.
-    if (layer === WEBPACK_LAYERS.server) {
-      // React needs to be bundled for Server Components so the special
-      // `react-server` export condition can be used.
-      if (reactPackagesRegex.test(request)) {
-        return
+    if (reactPackagesRegex.test(request)) {
+      switch (layer) {
+        case WEBPACK_LAYERS.server:
+          // We bundle all react packages in RSC
+          return
```

Comment:
Should we do the same for the action layer? Technically it should although I don't know about any case that uses React in a Server Action.

Reply:
Well the bundling is mostly to ensure we have separate instances. In server actions we never SSR the response so it doesn't have a conflicting environment which also need to load React. The main reason to bundle would be so we can control the conditions to get for instance the `react-server` condition. So maybe we should. but In practice it doesn't seem like this matters at the moment?